### PR TITLE
Mixer profile to open the posibility for vtol/mixed platform

### PR DESCRIPTION
**Pull request by shota3527**
I think this mixer_profile approach for VTOL requires the least amount of work for supporting VTOL. switching the PID/nav controllers between MC and FW.

### how to setup
https://github.com/shota3527/inav/blob/sh_mixer_profile/docs/MixerProfile.md

### known issues
- VTOL plane in multi-rotor mode can trigger false landing detection in RTH landing in a headwind. set `nav_disarm_on_landing ` to OFF might help
-  VTOL plane in multi-rotor mode struggles in windy conditions due to its physical limit

### updates in 202030725:
https://www.youtube.com/watch?v=6_4D3aLc8G8&t=204s
[INAV_7.0.0_cli_20230725_130613.txt](https://github.com/iNavFlight/inav/files/12206131/INAV_7.0.0_cli_20230725_130613.txt)

### old post:
It is still in development.
real flight on commit id:e0db9ae4c920530d405822f186f3b90f6aaddd2b
https://www.youtube.com/watch?v=9JEAN-SsOKw&t=2s
cli diff 
[INAV_6.0.0_cli_20221119_095038.txt](https://github.com/iNavFlight/inav/files/10045676/INAV_6.0.0_cli_20221119_095038.txt)

It is very experiential and it may cause a crash or flyaway or fly a servo
It should be ok unless mixer_profile other than 1 is used. May break a lot of things when performing a hot switch between profiles. reviews from other developers are appreciated

bench test
https://studio.youtube.com/video/aomNk
diff file used in video:
[INAV_6.0.0_cli_20221113_225555.txt](https://github.com/iNavFlight/inav/files/9997412/INAV_6.0.0_cli_20221113_225555.txt)

### CODE REVIEW

**Comment by shota3527** [comment]
Some big results recently, here is the cli file for t-1 ranger
[INAV_7.0.0_cli_20230725_130613.txt](https://github.com/iNavFlight/inav/files/12157449/INAV_7.0.0_cli_20230725_130613.txt)
also updated the documents
https://github.com/shota3527/inav/blob/sh_mixer_profile/docs/MixerProfile.md

**Comment by shota3527** [comment]
## It can use be used in sitl simulation now
## Convergence_Tilt_Tri_QuadPlane:

Realflight Model can be downloaded from here 
https://github.com/ArduPilot/SITL_Models/tree/master/RealFlight/Released_Models/QuadPlanes/Convergence_Tilt_Tri_QuadPlane
Diff file for quick setup:
[INAV_7.0.0_Tilt_Tri_cli_20230729_013947.txt](https://github.com/iNavFlight/inav/files/12196703/INAV_7.0.0_Tilt_Tri_cli_20230729_013947.txt)


```
./inav_7.0.0_SITL --sim=rf --chanmap=S01-01,S02-02,S03-03,S04-04,M01-05,M02-06,M03-07
```

**Comment by kasatka60** [comment]
Will there be a video about setting up inav?

**Comment by mmosca** [comment]
> Will there be a video about setting up inav?

For now, the document linked in the PR is the best we have.
It is still in active development and subject to changes, so a video could get out of date and is a lot more work to update than the documentation.

**Comment by macgyverek** [comment]
Can you add the same functionality to the another FC board - Speedybee F405 wing?

**Comment by shota3527** [comment]
> Can you add the same functionality to the another FC board - Speedybee F405 wing?

You can find it here by P-I-Engineer
https://github.com/P-I-Engineer/inav-piengr/tree/pi-shotamixer/src/main/target/SPEEDYBEEF405WING

sorry, I will not make target.c for another FC board because everyone prefers different pin mapping, I think It is better to leave the target.c to the manufacturer.

If you are not familiar with the compiling your firmware, i can send you a precompiled one

**Comment by P-I-Engineer** [comment]
https://www.youtube.com/watch?v=4X8yEgraxN4

File: src/main/drivers/pwm_mapping.c [hunk 1]
```diff
@@ -228,6 +228,20 @@ static void timerHardwareOverride(timerHardware_t * timer) {
     }
 }
 
+bool check_pwm_assigned_to_motor_or_servo(void)
```
Comment (DzikuVx) [inline src/main/drivers/pwm_mapping.c]: Please rename this functions to `camelCase`

File: src/main/flight/mixer_profile.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#ifndef MAX_MIXER_PROFILE_COUNT
+#define MAX_MIXER_PROFILE_COUNT 2
+#endif
+
+typedef struct mixerConfig_s {
+    int8_t motorDirectionInverted;
+    uint8_t platformType;
+    bool hasFlaps;
+    int16_t appliedMixerPreset;
+    uint8_t outputMode;
```
Comment (DzikuVx) [inline src/main/flight/mixer_profile.h]: We dropped this from mixerConfig so I think it can be removed

**Review by DzikuVx — CHANGES_REQUESTED** [review]
@shota3527 2 comments from my side. Plus, we will have to make some adjustments to Configurator to cope with the changes. And of course the removal of MOTOR_STOP what is good :)

File: src/main/drivers/pwm_mapping.c (hunk 1 above)
Comment (shota3527) [reply src/main/drivers/pwm_mapping.c]: Fixed. And  will delete this function once #9288 is merged

File: src/main/flight/mixer_profile.h (hunk 2 above)
Comment (shota3527) [reply src/main/flight/mixer_profile.h]: removed uint8_t outputMode;

**Comment by shota3527** [comment]
@DzikuVx 
bench tested and sim tested on commit [e4984a5](https://github.com/iNavFlight/inav/pull/8555/commits/e4984a52929d982d9dec1bed1105f238aaaf1b5d)
I have implemented the MSP to support the moved `motor stop feature`
Once configurator made changes on the `motor stop feature`, This PR should function same as master unless new feature from mixer_profie is used.
```
    case MSP2_INAV_MIXER:
        sbufWriteU8(dst, mixerConfig()->motorDirectionInverted);
        sbufWriteU8(dst, 0);
        sbufWriteU8(dst, mixerConfig()->motorstopOnLow);
        sbufWriteU8(dst, mixerConfig()->platformType);
        sbufWriteU8(dst, mixerConfig()->hasFlaps);
        sbufWriteU16(dst, mixerConfig()->appliedMixerPreset);
        sbufWriteU8(dst, MAX_SUPPORTED_MOTORS);
        sbufWriteU8(dst, MAX_SUPPORTED_SERVOS);
        break;
````

**Comment by shota3527** [comment]
corresponding configurator pull request 
https://github.com/iNavFlight/inav-configurator/pull/1829

**Review by DzikuVx — APPROVED** [review]

**Pull request merged**